examples/ota-requestor-app/linux/main.cpp

```diff
@@ -135,36 +237,50 @@ int main(int argc, char * argv[])
     err = mController.SetUdpListenPort(mStorage.GetListenPort());
     VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", chip::ErrorStr(err)));
 
-    err = mController.Init(initParams);
-    VerifyOrExit(err == CHIP_NO_ERROR, ChipLogError(Controller, "Init failure! Commissioner: %s", chip::ErrorStr(err)));
+    VerifyOrExit(rcac.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+    VerifyOrExit(noc.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+    VerifyOrExit(icac.Alloc(chip::Controller::kMaxCHIPDERCertLength), err = CHIP_ERROR_NO_MEMORY);
+
+    {
+        chip::MutableByteSpan nocSpan(noc.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan icacSpan(icac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+        chip::MutableByteSpan rcacSpan(rcac.Get(), chip::Controller::kMaxCHIPDERCertLength);
+
+        chip::Crypto::P256Keypair ephemeralKey;
+        SuccessOrExit(err = ephemeralKey.Initialize());
+
+        // TODO - OpCreds should only be generated for pairing command
+        //        store the credentials in persistent storage, and
+        //        generate when not available in the storage.
+        err = mOpCredsIssuer.GenerateNOCChainAfterValidation(mStorage.GetLocalNodeId(), 0, ephemeralKey.Pubkey(), rcacSpan,
+                                                             icacSpan, nocSpan);
```

Comment:
The app is self-generating credentials, rather than allowing some commissioning logic. This, like for chip-tool, works for development, but the assumptions that the example implementations of all blocks used here are what make "self" commissioning valid for this "app sample" need to be stated. All real requestors will need to be commissioned using realy means (and have been commissooned a-priori). I do understand why this example does this "expedited" version, but it should be extracted to a function/method like "ExampleSelfCommissioning" called from main, rather than fully inlined in main. This includes ALL the lines for mOpCredsIssuer, etc.

Reply:
Agreed. I'll move everything to its own function and add some documentation about why this is necessary (for now).